Scripts/BuildEngine.ps1

```diff
@@ -24,151 +24,160 @@
 #Requires -PSEdition Core
 
 param (
-    [Parameter(HelpMessage="System name to build, default to all")]
-    [ValidateSet('Windows', 'Linux', 'Darwin')]
-    [string[]] $SystemNames = @('Windows', 'Linux', 'Darwin'),
-
-    [Parameter(HelpMessage="Architecture type to build, default to x64")]
-    [ValidateSet('x64')]
-    [string] $Architectures = 'x64',
-
     [Parameter(HelpMessage="Configuration type to build")]
     [ValidateSet('Debug', 'Release')]
     [string[]] $Configurations = @('Debug', 'Release'),
 
     [Parameter(HelpMessage="Whether to run build, default to True")]
-    [bool] $RunBuilds = $True
+    [bool] $RunBuilds = $True,
+
+    [Parameter(HelpMessage="VS version use to build, default to 2019")]
+    [ValidateSet(2019, 2022)]
+    [int] $VSVersion = 2019
 )
 
+
+##Define None variables
+Set-Variable ARCHITECTURE -Option None -value 'x64'
+Set-Variable SYSTEM_NAMES -Option None -value 'Windows', 'Linux', 'Darwin'
+
+# # CMake arguments variables
+
+Set-Variable CMAKE_GENERATOR_DARWIN -Option None -value 'Xcode'
```

Comment:
Since we can detect the OS system, I think we can use a single variable `CMAKE_GENERATOR` and define its value depending on whether we are on Win/Mac/Linux